### PR TITLE
sig-cloud-provider: Consolidate k/k cloud provider test jobs

### PR DESCRIPTION
**Pull request by justaugustus**
- sig-cloud-provider: Consolidate k/k cloud provider test jobs
- sig-cloud-provider: Update job config OWNERS

(For my own sanity, we can look at job configs/testgrid annotations in a separate PR or leave that to the individual OWNERS.)

Part of cloud provider SIG consolidation: https://github.com/kubernetes/community/issues/2770#issuecomment-516306574

### CODE REVIEW

**Comment by justaugustus** [comment]
Looks like the security jobs needed to be regenerated:
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/test-infra/13703/pull-test-infra-bazel/1156457712029011968

Opened #13704 for this.

**Comment by spiffxp** [comment]
You need to regenerate the jobs as part of this PR, see https://github.com/kubernetes/test-infra/tree/master/config/jobs#adding-or-updating-jobs

File: config/jobs/kubernetes/cloud-provider-azure/OWNERS [hunk 1]
```diff
@@ -7,3 +7,6 @@ approvers:
 - justaugustus
 - karataliu
 - khenidak
+
+labels:
+- area/provider/azure
```
Comment (spiffxp) [inline config/jobs/kubernetes/cloud-provider-azure/OWNERS]: this label doesn't exit

File: config/jobs/kubernetes/sig-cloud-provider/aws/OWNERS [hunk 2]
```diff
@@ -4,3 +4,6 @@ approvers:
 - chrislovecnm
 - justinsb
 - shyamjvs
+
+labels:
+- area/provider/aws
```
Comment (spiffxp) [inline config/jobs/kubernetes/sig-cloud-provider/aws/OWNERS]: this label doesn't exist

File: config/jobs/kubernetes/sig-cloud-provider/gcp/OWNERS [hunk 3]
```diff
@@ -3,3 +3,6 @@
 approvers:
 - abgworrall
 - dashpole
+
+labels:
+- area/provider/gcp
```
Comment (spiffxp) [inline config/jobs/kubernetes/sig-cloud-provider/gcp/OWNERS]: this label doesn't exist

**Comment by justaugustus** [comment]
@spiffxp -- thanks for the heads up on the job generation! Updated.
Labels are updated in #13677, if you've got a moment to review that one.

**Comment by spiffxp** [comment]
Requested some changes on https://github.com/kubernetes/test-infra/pull/13677 merges, we can `/hold cancel` this once that merges and takes effect

**Comment by justaugustus** [comment]
@spiffxp @andrewsykim -- This is good to go now that #13677 has merged and the new cloud provider labels are active.

(Noting that I didn't attempt to rename the job config files as people may have opinions on what to name their configs and I didn't want to bikeshed here.)

**Pull request merged**